### DISCUSSION TITLE
How to implement docker with prisma?

### DISCUSSION THREAD

**Discussion by tika** [prisma/prisma#16112] [Q&A] [answered]
Here's my Dockerfile:
```
FROM node:18-alpine as base

WORKDIR /home/node/app

COPY package.json yarn.lock ./

RUN yarn install

COPY tsconfig.json ./
COPY src ./src
COPY .env ./
COPY prisma ./prisma

RUN yarn prisma generate

# Add any additional env files etc. here


FROM base as production

ENV NODE_PATH=./dist

RUN yarn build
```
I keep getting this error: 
```
 => ERROR [base 9/9] RUN yarn prisma generate                                   0.7s
------
 > [base 9/9] RUN yarn prisma generate:
#13 0.253 yarn run v1.22.19
#13 0.285 $ /home/node/app/node_modules/.bin/prisma generate
#13 0.696 Error: Unknown binaryTarget linux-arm64-openssl-undefined and no custom engine files were provided
#13 0.713 error Command failed with exit code 1.
#13 0.713 info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
------
executor failed running [/bin/sh -c yarn prisma generate]: exit code: 1
ERROR: Service 't3' failed to build : Build failed
```

Here's my docker-compose.yml
```
version: "3.4"
services:
  postgres:
    image: postgres:latest
    container_name: t3-postgres
    hostname: postgres
    ports:
      - 5432:5432
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: Todo
    volumes:
      - postgres-data:/var/lib/postgresql/data
    restart: unless-stopped
  t3:
    build:
      context: .
      dockerfile: Dockerfile
      target: base
    volumes:
      - ./src:/home/node/app/src
      - ./nodemon.json:/home/node/app/nodemon.json
    container_name: t3-fastify-prisma
    expose:
      - "4000"
    ports:
      - "4000:4000"
    environment:
      DATABASE_URL: "postgresql://postgres:postgres@db:5432/mydb?schema=public"
    command: npm run dev
    depends_on:
      - postgres

volumes:
  postgres-data:
```

Let me know if you need anything else.

**Comment by pantharshit00** [comment, marked as answer]
Are you using M1 Mac? If yes, then this is related to https://github.com/prisma/prisma/issues/8478

You can try adding openssl once though in the image by adding `RUN apk update && apk add nodejs npm openssl python g++ make && rm -rf /var/cache/apk/*`, otherwise I will suggest you to use `node:18-slim` image.

**Reply by pantharshit00** [reply]
Hello,

Is this still relevant to you? Please let me know if the above steps resolves the issue or not.

**Reply by tika** [reply]
Edit: Yes I did use this solution. Thank you for the help!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
